### PR TITLE
Feature/234: GET main flow of a partitioning

### LINKED ISSUE #234: GET /partitionings/{partId}/main-flow
## Feature
Implement `GET /partitionings/{partId}/main-flow` V2 REST API endpoint on the server

It returns the main flow of the given partitioning.

## Proposed Solution
```scala
protected val getPartitioningMainFlowEndpointV2
    : PublicEndpoint[Long, ErrorResponse, SingleSuccessResponse[FlowDTO], Any] = {
    apiV2.get
      .in(V2Paths.Partitionings / path[Long]("partitioningId") / V2Paths.MainFlow)
      .out(statusCode(StatusCode.Ok))
      .out(jsonBody[SingleSuccessResponse[FlowDTO]])
  }
```

### PR DESCRIPTION
**Pull request by lsulak**
Closes #234 

* DB function
* Balta-based DB tests
* Endpoint, i.e. Server part of this functionality
* 50-59 DB errors now have its representative err in HTTP / Scala layer

### CODE REVIEW

**Comment by lsulak** [comment]
Release Notes:
* Introduces GET endpoint for retrieval of Main Flow of a given partitioning

File: database/src/main/postgres/runs/V1.9.9__get_partitioning_main_flow.sql [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    SELECT F.id_flow,
+           F.flow_name,
+           F.flow_description,
+           F.from_pattern
+    FROM flows.flows AS F
+    WHERE F.fk_primary_partitioning = i_partitioning_id
+    INTO id_flow, flow_name, flow_description, from_pattern;
+
+    IF NOT FOUND THEN
```
Comment (benedeki) [inline database/src/main/postgres/runs/V1.9.9__get_partitioning_main_flow.sql:69]: Should never happen, but fine to have it 👍

File: server/src/main/scala/za/co/absa/atum/server/api/exception/DatabaseError.scala [hunk 2]
```diff
@@ -23,5 +23,6 @@ object DatabaseError {
   case class GeneralDatabaseError(message: String) extends DatabaseError
   case class ConflictDatabaseError(message: String) extends DatabaseError
   case class NotFoundDatabaseError(message: String) extends DatabaseError
+  case class ErrorInDataDatabaseError(message: String) extends DatabaseError
```
Comment (benedeki) [inline server/src/main/scala/za/co/absa/atum/server/api/exception/DatabaseError.scala:26]: **Non-consequential for this PR:**
What if `DatabaseError` would have `status`, and `statusText` fields?

**Review by benedeki — COMMENTED** [review]
- [x] code reviewed
- [ ] pulled
- [ ] built
- [ ] run

**Review by benedeki — DISMISSED** [review]

File: server/src/main/scala/za/co/absa/atum/server/api/exception/DatabaseError.scala (hunk 2 above)
Comment (salamonpavel) [reply server/src/main/scala/za/co/absa/atum/server/api/exception/DatabaseError.scala:26]: GeneralDatabaseError is used to represent all kinds of exceptions (not only fa-db exceptions); therefore we don't have any status and status test.

File: server/src/main/scala/za/co/absa/atum/server/api/database/runs/functions/GetPartitioningMainFlow.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import za.co.absa.db.fadb.doobie.DoobieEngine
+import za.co.absa.db.fadb.doobie.DoobieFunction.DoobieSingleResultFunctionWithStatus
+import za.co.absa.db.fadb.status.aggregation.implementations.ByFirstErrorStatusAggregator
+import za.co.absa.db.fadb.status.handling.implementations.StandardStatusHandling
+import zio.{Task, URLayer, ZIO, ZLayer}
+
+class GetPartitioningMainFlow(implicit schema: DBSchema, dbEngine: DoobieEngine[Task])
+    extends DoobieSingleResultFunctionWithStatus[Long, Option[FlowDTO], Task](
+      input => Seq(fr"$input"),
+      Some("get_partitioning_main_flow")
```
Comment (salamonpavel) [inline server/src/main/scala/za/co/absa/atum/server/api/database/runs/functions/GetPartitioningMainFlow.scala:33]: why the override?

File: server/src/main/scala/za/co/absa/atum/server/api/exception/DatabaseError.scala (hunk 2 above)
Comment (benedeki) [reply server/src/main/scala/za/co/absa/atum/server/api/exception/DatabaseError.scala:26]: Good point.

File: server/src/main/scala/za/co/absa/atum/server/api/database/runs/functions/GetPartitioningMainFlow.scala (hunk 3 above)
Comment (lsulak) [reply server/src/main/scala/za/co/absa/atum/server/api/database/runs/functions/GetPartitioningMainFlow.scala:33]: You are right, not needed at all, I will change it

**Review by salamonpavel — APPROVED** [review]

**Review by benedeki — APPROVED** [review]

**Review by salamonpavel — DISMISSED** [review]

**Review by salamonpavel — APPROVED** [review]

**Pull request merged**